pkg/cluster/internal/providers/docker/network.go

```diff
@@ -50,11 +50,28 @@ func ensureNetwork(name string, ipFamily config.ClusterIPFamily) error {
 	if string(out) == name+"\n" {
 		return nil
 	}
-	// TODO: ipv6 subnet should probably not be fixed
-	// Though maybe just require the user to handle this by creating the network
-	// as they desire before running kind ...
-	if ipFamily == config.IPv6Family {
-		return exec.Command("docker", "network", "create", "-d=bridge", "--ipv6", "--subnet=fc00:db8:2::/64", name).Run()
+
+	// generate unique subnet per network based on the name
+	// obtained from the ULA fc00::/8 range
+	subnet := generateULASubnetFromName(name)
+
+	return exec.Command("docker", "network", "create", "-d=bridge", "--ipv6", "--subnet", subnet, name).Run()
+}
+
+// generateULASubnetFromName generate an IPvs subnet based on the name passed as parameter
```

Comment:
typo `IPv6`
i'd move the comments above the call to here and clarify a little more why this is needed.